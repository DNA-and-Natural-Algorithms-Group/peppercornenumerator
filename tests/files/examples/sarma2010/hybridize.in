# This is a test of the resting state / transient state dichotomy during enumeration.
# Without it, hybridizing strands with two long domains will lead to enumeration of infinite polymers.
# However, successfully, no infinite polymers are formed.
domain 1 : long
domain 2 : short
domain 3 : short
domain 4 : long
domain 5 : short

# a strands refer to blue "top" strands, labeled left to right
strand a1 : 1 4* 3
strand a2 : 1 4*
strand a3 : 5 1
strand a4 : 5 1 3

# b strands refer to bottom strands, labeled left to right
strand b1 : 3* 4 1* 1*
strand b2 : 4 1* 5* 4

complex G1top :
a2
..

complex G1bot :
b1
....

complex G1 :
a2 b1
(( + .)).

