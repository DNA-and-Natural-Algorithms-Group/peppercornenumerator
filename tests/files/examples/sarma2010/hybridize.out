###### Enumerated Output ######


# Domains 
sequence 1 = : 12
sequence 2 = : 6
sequence 3 = : 6
sequence 4 = : 12
sequence 5 = : 6


# End-state Complexes 
G1top
1 4*
..

4
1 4* + 1 4* + 3* 4 1* 1*
(.+((+.)))

G1
1 4* + 3* 4 1* 1*
((+.)).

G1bot
3* 4 1* 1*
....

###############################


# Fast (Transition) Complexes 
8
1 4* + 3* 4 1* 1*
.(+.)..

6
1 4* + 3* 4 1* 1*
(.+..).

7
1 4* + 3* 4 1* 1*
(.+...)

14
1 4* + 3* 4 1* 1*
((+.).)

9
1 4* + 3* 4 1* 1* + 1 4* + 3* 4 1* 1*
((+.))(+)(+.)..

###############################


# Reactions 
8 -> G1 
8 -> 14 
6 -> G1 
7 -> 14 
G1top + G1 -> 4 
G1top + G1bot -> 8 
G1top + G1bot -> 6 
G1top + G1bot -> 7 
4 + G1bot -> 9 
14 -> G1 
9 -> G1 + G1 
###############################


# Condensed Reactions 
G1top + G1 -> 4 
G1top + G1bot -> G1 
4 + G1bot -> G1 + G1 
